app/src/main/java/de/westnordost/streetcomplete/MainActivity.java

```diff
@@ -247,6 +252,36 @@ public void onServiceDisconnected(ComponentName className)
 		{
 			questController.deleteOld();
 		}
+
+		Intent intent = getIntent();
+		Uri data = intent.getData();
+		if (Intent.ACTION_VIEW.equals(intent.getAction()) && data != null)
+		{
+			if ("geo".equals(data.getScheme()))
+			{
+				double latitude = 0;
+				double longitude = 0;
+				float zoom = -1;
+
+				String geoUriRegex = "^geo:(-?[0-9]*\\.?[0-9]+),(-?[0-9]*\\.?[0-9]+).*?(?:\\?z=([0-9]*\\.?[0-9]+))?";
+				Pattern pattern = Pattern.compile(geoUriRegex);
+				Matcher matcher = pattern.matcher(data.toString());
+				while (matcher.find())
```

Comment:
Why while?

Reply:
By using [`find()`](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Matcher.html#find--) it is also possible to extract the result from a given substring. I agree, that this doesn't make that much sense in our use case, so the following is enough:

```suggestion
				if (matcher.matches())
				{
				        latitude = Double.parseDouble(matcher.group(1));
					longitude = Double.parseDouble(matcher.group(2));
					if (matcher.group(3) != null) {
						zoom = Float.valueOf(matcher.group(3));
					}
				}
```